frontend/src/components/pages/CourseDashboard.js

```diff
@@ -24,8 +68,46 @@ export function CourseDashboard() {
             .catch((error) => setError(error.message));
     }, []);
 
+    useEffect(() => {
+        const userId = Cookies.get('userId');
+
+        if (!userId) {
+            setError('User ID not found');
+            return;
+        }
+
+        fetch(`http://localhost:4000/user/${userId}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then((data) => setUserInfo(data.user))
+            .catch((error) => setError(error.message));
+    }, []);
+
     if (error) return <p>Error: {error}</p>;
     if (!courseInfo) return <p>Loading...</p>;
```

Comment:
Could be combine both of them? What do you think?

```js
if (!courseInfo || !userInfo) return <p>Loading...</p>;
```

Maybe even create Spinner (or get a Spinner loader icon or something? could be for later in another PR)